### PR TITLE
fix npm deprecation warnings

### PR DESCRIPTION
**Pull request by ETLaurent**
[PRO-2560](https://linear.app/apostrophecms/issue/PRO-2560/as-a-new-a3-developer-i-should-see-fewer-deprecation-warnings-when-i)

## Summary

Fix most of the deprecation warnings for which we have an impact directly in the apostrophe `package`:

**before**

```sh
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated eslint-loader@4.0.2: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated core-js@1.2.7: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @stylelint/postcss-markdown@0.36.2: Use the original unforked package instead: postcss-markdown
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older ve
```

**after**

```sh
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
```

## What are the specific steps to test this change?

Run:
- `rm -rf package-lock.json node_modules`
- `npm install`

You should see the "**after**" deprecation warnings ☝️ 

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [x] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: .stylelintrc
Comment (ETLaurent) [inline .stylelintrc:93]: see
- https://stylelint.io/migration-guide/to-14/
- https://github.com/postcss/postcss#syntaxes

File: modules/@apostrophecms/asset/lib/webpack/apos/webpack.config.js
Comment (ETLaurent) [inline modules/@apostrophecms/asset/lib/webpack/apos/webpack.config.js:54]: revert and fix following error to be able to link apostrophe again:

```sh
[nodemon] restarting due to changes...
[nodemon] node_modules/apostrophe/modules/@apostrophecms/any-doc-type/index.js

[nodemon] starting `node app.js`
[nodemon] forking
[nodemon] child pid: 35019
internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module 'glob'
Require stack:
- /Users/etienne/Sites/apostrophe/apostrophe/index.js
- /Users/etienne/Sites/apostrophe/testbed/app.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at Object.<anonymous> (/Users/etienne/Sites/apostrophe/apostrophe/index.js:10:14)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/etienne/Sites/apostrophe/apostrophe/index.js',
    '/Users/etienne/Sites/apostrophe/testbed/app.js'
  ]
}
[nodemon] app crashed - waiting for file changes before starting...
```

File: modules/@apostrophecms/asset/lib/webpack/src/webpack.config.js
Comment (ETLaurent) [inline modules/@apostrophecms/asset/lib/webpack/src/webpack.config.js:65]: see https://github.com/apostrophecms/apostrophe/pull/3700/files#r839570205

File: package.json
Comment (ETLaurent) [inline package.json:93]: see https://github.com/apostrophecms/apostrophe/pull/3700/files#r839568970

File: package.json
Comment (ETLaurent) [inline package.json:105]: using `stylelint@14` fixes `@stylelint/postcss-markdown` warning

File: package.json
Comment (ETLaurent) [inline package.json:109]: `stylelint-*` dependencies must be updated as well, to match the `stylelint@14` version in their peer dependencies

File: package.json
Comment (ETLaurent) [reply package.json:105]: **before**

```sh
➜  apostrophe git:(2560-npm-warnings) ✗ npx stylelint modules/**/*.{scss,vue} 

modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarLocale.vue
 312:3  ✖  Expected variable, function or keyword for "10px" of "font-size"   scale-unlimited/declaration-strict-value
 313:3  ✖  Expected "color" to come before "font-size"                        order/properties-order                  

modules/@apostrophecms/login/ui/apos/components/TheAposLoginHeader.vue
  92:5   ✖  Expected variable, function or keyword for "#F8F9FA" of "color"    scale-unlimited/declaration-strict-value
  95:18  ✖  Expected a trailing semicolon                                      declaration-block-trailing-semicolon    
  99:7   ✖  Expected variable, function or keyword for "21px" of "font-size"   scale-unlimited/declaration-strict-value
 101:5   ✖  Unexpected empty line before closing brace                         block-closing-brace-empty-line-before   

modules/@apostrophecms/ui/ui/apos/components/AposSpinner.vue
 113:5  ✖  Unexpected vendor-prefix "-webkit-transform"   property-no-vendor-prefix
 115:5  ✖  Expected indentation of 2 spaces               indentation              
 116:5  ✖  Unexpected vendor-prefix "-webkit-transform"   property-no-vendor-prefix
 122:5  ✖  Unexpected vendor-prefix "-webkit-transform"   property-no-vendor-prefix
 124:5  ✖  Expected indentation of 2 spaces               indentation              
 125:5  ✖  Unexpected vendor-prefix "-webkit-transform"   property-no-vendor-prefix
```

**after**

```sh
➜  apostrophe git:(2560-npm-warnings) npx stylelint modules/**/*.{scss,vue} 

modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarLocale.vue
 312:3  ✖  Expected variable, function or keyword for "10px" of "font-size"  scale-unlimited/declaration-strict-value
 313:3  ✖  Expected "color" to come before "font-size"                       order/properties-order                  

modules/@apostrophecms/login/ui/apos/components/TheAposLoginHeader.vue
  92:5   ✖  Expected variable, function or keyword for "#F8F9FA" of "color"   scale-unlimited/declaration-strict-value
  95:18  ✖  Expected a trailing semicolon                                     declaration-block-trailing-semicolon    
  99:7   ✖  Expected variable, function or keyword for "21px" of "font-size"  scale-unlimited/declaration-strict-value
 101:5   ✖  Unexpected empty line before closing brace                        block-closing-brace-empty-line-before   

modules/@apostrophecms/ui/ui/apos/components/AposSpinner.vue
 113:5  ✖  Unexpected vendor-prefix "-webkit-transform"  property-no-vendor-prefix
 115:5  ✖  Expected indentation of 2 spaces              indentation              
 116:5  ✖  Unexpected vendor-prefix "-webkit-transform"  property-no-vendor-prefix
 122:5  ✖  Unexpected vendor-prefix "-webkit-transform"  property-no-vendor-prefix
 124:5  ✖  Expected indentation of 2 spaces              indentation              
 125:5  ✖  Unexpected vendor-prefix "-webkit-transform"  property-no-vendor-prefix
```

Comment (ETLaurent) [reply package.json:105]: ☝️  no change then, but I still have to figure out when is the stylelint run, if run at all

**Comment by ETLaurent** [comment]
at this point:

```sh
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated eslint-loader@4.0.2: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
```

**Comment by ETLaurent** [comment]
at this point:

```sh
npm WARN deprecated eslint-loader@4.0.2: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
```

**Comment by ETLaurent** [comment]
`request` is used by the following packages:

```
apostrophe@3.16.1 /Users/etienne/Sites/apostrophe/apostrophe
├─┬ oembetter@1.0.1
│ └── request@2.88.2 
└─┬ uploadfs@1.19.0
  ├─┬ azure-storage@2.10.7
  │ └── request@2.88.2  deduped
  └── request@2.88.2  deduped
```

We have the possibility to change `request` on `oembetter` and `uploadfs` packages, and we can change `azure-storage` to the new following packages, that do not include `request` (`@azure/core-http` uses `node-fetch`):

```
azure-storage@2.10.7 | Apache-2.0 | deps: 12 | versions: 50
Microsoft Azure Storage Client Library for Node.js
http://github.com/Azure/azure-storage-node

DEPRECATED ⚠️  - Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
```

An issue should be assigned for this work.

File: package.json
Comment (ETLaurent) [inline package.json:127]: fixes `npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.` warning.

see https://www.npmjs.com/package/@babel/eslint-parser

**Comment by ETLaurent** [comment]
`popper.js` is used by `v-tooltip`:

```
apostrophe@3.16.1 /Users/etienne/Sites/apostrophe/apostrophe
└─┬ v-tooltip@2.1.3
  └── popper.js@1.16.1
```

The solution to use the new Popper v2 at @popperjs/core would be to upgrade `v-tooltip` to the `next` tag (`v-tooltip@3.0.0-beta.20`), which support vue 2 and vue 3 (see https://github.com/Akryum/floating-vue/releases)

We may want to way for this package to be stable

**Comment by ETLaurent** [comment]
`uuid` is used by:

```
apostrophe@3.16.1 /Users/etienne/Sites/apostrophe/apostrophe
├─┬ nyc@15.1.0
│ └─┬ istanbul-lib-processinfo@2.0.2
│   └── uuid@3.4.0 
├─┬ oembetter@1.0.1
│ └─┬ request@2.88.2
│   └── uuid@3.4.0  deduped
└─┬ uploadfs@1.19.0
  ├─┬ @google-cloud/storage@5.18.3
  │ └─┬ @google-cloud/common@3.10.0
  │   └─┬ teeny-request@7.1.3
  │     └── uuid@8.3.2 
  ├─┬ aws-sdk@2.1104.0
  │ └── uuid@3.3.2 
  └─┬ azure-storage@2.10.7
    └── uuid@3.4.0  deduped
```

The work of upgrading this package to version 7 or higher is significant, and might be done separately.

**Comment by ETLaurent** [comment]
`har-validator` is included by `oembetter`, via `request`:

```
apostrophe@3.16.1
└─┬ oembetter@1.0.1
  └─┬ request@2.88.2
    └── har-validator@5.1.5
```

Once we do [that](https://github.com/apostrophecms/apostrophe/pull/3700#issuecomment-1084599704), problem will be fixed

File: package.json
Comment (ETLaurent) [inline package.json:128]: Deprecated and unused.
Fixes the `npm WARN deprecated eslint-loader@4.0.2: This loader has been deprecated. Please use eslint-webpack-plugin` warning

see https://github.com/webpack-contrib/eslint-loader

**Comment by ETLaurent** [comment]
**what we now have**

```sh
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated azure-storage@2.10.7: Please note: newer packages @azure/storage-blob, @azure/storage-queue and @azure/storage-file are available as of November 2019 and @azure/data-tables is available as of June 2021. While the legacy azure-storage package will continue to receive critical bug fixes, we strongly encourage you to upgrade. Migration guide can be found: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/MigrationGuide.md
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
```

File: test/assets.js
Comment (ETLaurent) [inline test/assets.js:161]: fixes timeout

Comment (ETLaurent) [reply test/assets.js:161]: cc @ValJed

**Comment by boutell** [comment]
The transition to `@azure/core-http` will take some doing, so we should take a look and see how hard it would be to just use `node-fetch` ourselves with Azure. My reason for suggesting this is that npm modules are easy to constantly iterate and release new breaking major versions of, causing extra work for us, but REST APIs are very hard to deprecate. Azure will not break our code as soon if it speaks directly to their REST API, and this is usually not very hard to do. Usually...

**Comment by jean-smaug** [comment]
🥐

**Comment by ETLaurent** [comment]
> The transition to `@azure/core-http` will take some doing, so we should take a look and see how hard it would be to just use `node-fetch` ourselves with Azure. My reason for suggesting this is that npm modules are easy to constantly iterate and release new breaking major versions of, causing extra work for us, but REST APIs are very hard to deprecate. Azure will not break our code as soon if it speaks directly to their REST API, and this is usually not very hard to do. Usually...

@boutell what do you suggest we do? First merge this PR and see about azure and node-fetch in a separate one?

**Review by ValJed — APPROVED** [review]

File: package.json
Comment (boutell) [reply package.json:105]: This should become a separate ticket.

**Review by boutell — APPROVED** [review]
Good to approve and merge this work incrementally, if there is a stylelint issue with actually running that please open a separate ticket, and continue the uploadfs work in a separate ticket for sure

File: package.json
Comment (ETLaurent) [reply package.json:105]: @boutell created: https://linear.app/apostrophecms/issue/PRO-2735/fix-stylelint-issues

**Pull request merged**
